### DISCUSSION TITLE
How to Remove Yourself as a Co-Author from a Commit?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#76660] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I recently found that I was co-authored in a commit without my knowledge or consent. Since I don't own the repository, I'm seeking guidance on the proper steps to remove myself as a co-author from the commit.

**_Shouldn't there be a verification step before co-authoring in the commit?_**

**Comment by deeptejd** [comment]
Some repositories have contribution guidelines that specify how contributions should be made and attributed. Check if the repository has such guidelines, and if the addition of co-authors without consent is addressed.
Reach out to the person who added you as a co-author.
If the contributor is unresponsive or unwilling to make the change, you can open an issue or pull request on the repository explaining the situation.
If the above steps do not resolve the issue, consider reaching out to the maintainers of the repository. Explain the situation and ask for their assistance in correcting the attribution.

**Reply by ghost** [reply]
Would that remove the commit link from that pair's extraordinary badge? I got that badge for that commit. I don't want that link in my badge and the badge itself for the commit in which I have no contribution.

**Reply by deeptejd** [reply]
No, it would not, but you can hide the badge from your profile

**Reply by ghost** [reply]
Yeah, it did not, and now I have to hide the badge for this nonsensical mistake. I can't even display my badge for my contribution afterward. This is absurd.

**Comment by Sarath-Kumar-G** [comment]
Removing yourself as a co-author from a commit in Git, a version control system, can be a bit complex, especially if the commit has already been pushed to a remote repository. 

**- Rewriting History Locally:**
If the commit is still local (not pushed to a remote repository), you can use the git rebase command to rewrite history. This involves editing the commit message to remove your name as a co-author.

First, identify the commit hash you want to edit using the `git log`.
Then, start an interactive rebase with `git rebase -i <commit_hash>`. Replace <commit_hash> with the hash of the commit just before the one you want to edit.
In your editor that opens, change the word pick to edit next to the commit you want to modify, then save and close the editor.
Once the rebase has started and stopped at your commit, use `git commit --amend` to edit the commit message. **Remove your name as a co-author and save the changes.**
Finally, continue the rebase with `git rebase --continue`.

**-Force Pushing Changes:**
If the commit has been pushed to a remote repository, you'll need to force-push the changes after rewriting the history. This can be disruptive to others working on the same repository, so it's important to coordinate with your team.

After completing the local history rewrite, use `git push origin <branch_name> --force` to push the changes to the remote repository. Replace <branch_name> with the name of your branch. Thanks

**Reply by ghost** [reply]
Would that remove the commit link from that pair's extraordinary badge? I got that badge for that commit.

**Reply by Sarath-Kumar-G** [reply]
It should remove the commit.

**Reply by helensilva14** [reply]
Hi @Sarath-Kumar-G, these steps worked for my scenario, thank you very much for sharing!

**Comment by jackyyy65** [comment]
Sorry to hear that you were co-authored in a commit without your consent. To remove yourself as a co-author, communicate your concerns to the repository owner/administrator and request to be removed. Ideally, there should be a verification [step](https://doesitsnow.net/) to ensure consent before adding co-authors. Open communication and seeking resolution through the repository owner or support team can help address the issue. Good luck in resolving this matter!
